### PR TITLE
feat: Edge sync indicator

### PR DESCRIPTION
**Pull request by dmaretskyi**
image
image

### CODE REVIEW

File: packages/plugins/plugin-space/src/components/SyncStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    });
+  }, []);
+
+  const needsToUpload = state.missingOnRemote > 0 || state.differentDocuments > 0;
+  const needsToDownload = state.missingOnLocal > 0 || state.differentDocuments > 0;
+  const notConnectedToEdge = state.spacesNotConnectedToEdge.size > 0;
+
+  return (
+    <StatusBar.Item title={JSON.stringify(state, null, 2)}>
+      {notConnectedToEdge ? (
```
Comment (richburdon) [inline packages/plugins/plugin-space/src/components/SyncStatus.tsx:33]:
```tsx
import { Icon } from 'react-ui';
<Icon icon={'ph--cloud-slash--regular' || ...} size={4} />
```

File: packages/plugins/plugin-space/src/components/SyncStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      setState(state);
+    });
+  }, []);
+
+  const needsToUpload = state.missingOnRemote > 0 || state.differentDocuments > 0;
+  const needsToDownload = state.missingOnLocal > 0 || state.differentDocuments > 0;
+  const notConnectedToEdge = state.spacesNotConnectedToEdge.size > 0;
+
+  return (
+    <StatusBar.Item title={JSON.stringify(state, null, 2)}>
```
Comment (richburdon) [inline packages/plugins/plugin-space/src/components/SyncStatus.tsx:32]: TODO(burdon): Click for popup status

File: packages/plugins/plugin-space/src/components/SyncStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  install(client.spaces.get());
+
+  return () => {
+    for (const unsubscribe of Object.values(unsubscribeCallbacks)) {
+      unsubscribe();
+    }
+  };
+};
+
+type EdgeSyncState = {
```
Comment (richburdon) [inline packages/plugins/plugin-space/src/components/SyncStatus.tsx:74]: is there a better place to put all of the edge deps? (outside of ux?)

**Review by richburdon — APPROVED** [review]

File: packages/plugins/plugin-space/src/components/SyncStatus.tsx (hunk 3 above)
Comment (dmaretskyi) [reply packages/plugins/plugin-space/src/components/SyncStatus.tsx:74]: so its not in plugin-space?

yeah, probably we should make an edge plugin

**Pull request merged**